base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1888,97 +1892,159 @@ where
+                    // ----------------------------------------------------------------------------
+                    // simple one-sided address
+                    [Opcode::PushPubKey(spending_key_public)] => {
+                        let keys = match self.resources.db.get_all_known_one_sided_payment_scripts() {
+                            Ok(keys) => keys,
+                            Err(_) => return None,
+                        };
+
+                        match keys
+                            .into_iter()
+                            .find(|x| PublicKey::from_secret_key(&x.private_key) == **spending_key_public)
+                        {
+                            None => None,
+                            Some(x) => {
+                                match PrivateKey::from_bytes(
+                                    CommsPublicKey::shared_secret(&x.private_key, &output.sender_offset_public_key)
+                                        .as_bytes(),
+                                ) {
+                                    Ok(spend_key) => Some((output.clone(), x.private_key.clone(), spend_key)),
+                                    Err(_) => None,
+                                }
                             },
                         }
-                        trace!(
-                            target: LOG_TARGET,
-                            "One-sided payment Output {} with value {} recovered",
-                            output_hex,
-                            committed_value,
-                        );
+                    },
+
+                    // ----------------------------------------------------------------------------
+                    // one-sided stealth address
+                    // NOTE: Extracting the nonce R and a spending (public aka scan_key) key from the script
+                    // NOTE: [RFC 203 on Stealth Addresses](https://rfc.tari.com/RFC-0203_StealthAddresses.html)
+                    [Opcode::PushPubKey(nonce), Opcode::Drop, Opcode::PushPubKey(provided_spending_public)] => {
+                        let secret_key = self.node_identity.secret_key().clone();
+                        let public_key = self.node_identity.public_key();
+
+                        // computing shared secret
+                        let c = RistrettoSecretKey::from_bytes(
+                            DomainSeparatedHasher::<Blake256, GenericHashDomain>::new("stealth address")
+                                .chain(PublicKey::shared_secret(&secret_key, nonce.as_ref()).as_bytes())
+                                .finalize()
+                                .as_ref(),
+                        )
+                        .unwrap();
+
+                        // calculating spending (public) key
+                        let ks = PublicKey::from_secret_key(&c) + public_key;
+
+                        if &ks != provided_spending_public.as_ref() {
+                            return None;
```

Comment:
I don't think return here will give the correct behaviour. 
If this case here is not correct, we should not drop all other correct UTXO and return back with none. This is in the middle of an iteration, I suspect you only want to continue here?

Reply:
Yes, `return None` is to skip, to return from `filter_map` closure.  This function is to only find and add certain outputs, it's not supposed to pass anything through.

Reply:
I was also thrown by this. I think the filter_map is too large and you should probably rather use a simple for loop